pkg/network/event_common.go

```diff
@@ -505,3 +480,13 @@ func (s StatCounters) Max(other StatCounters) StatCounters {
 		TCPEstablished: maxUint32(s.TCPEstablished, other.TCPEstablished),
 	}
 }
+
+// isUnderflow checks if a metric has "underflowed", i.e.
+// the most recent value is less than what was seen
+// previously. We distinguish between an "underflow" and
+// an integer overflow if the change is greater than
+// some preset max value; if the change is greater, then
+// its an underflow
+func isUnderflow(previous, current, maxChange uint64) bool {
+	return current < previous && (current-previous) > maxChange
```

Comment:
should it be `previous-current` since previous is bigger?

Reply:
Even if `current < previous`, the subtraction will have the correct delta, and if this is the case of an overflow, we want that value.

Reply:
I am likely missing something here. We are doing the subtraction only in the case when `current` is less than `previous`. My understanding is that doing `current-previous` will always be a negative number and because of being unsigned, will be a big number that does not really mean anything with respect to maximum byte count. Relevant: https://go.dev/play/p/KU6ia7wlPC__J
If this is indeed the correct way, it might be helpful to add a comment explaining why.

Reply:
Go handles unsigned overflows. So if `previous` was `math.MaxUint64`, and `current` was `1`, e.g., `current-previous` would be `2`. The logic here is handling the case of if we see a future value that is lower, is it valid? The only way to tell the difference between a genuine overflow and a "bogus" value is to set some limits. So if the increment `current-previous` is above a certain threshold (that we consider a valid change), we consider an underflow, a valid overflow otherwise.

Reply:
Thanks for the explanation. Was somehow confused by all the unsigned arithmetic.